app/controllers/foreman_statistics/api/v2/statistics_controller.rb

```diff
@@ -27,6 +28,13 @@ def index
                             :model_count => @model_count, :mem_size => @mem_size, :mem_free => @mem_free,
                             :swap_free => @swap_free, :mem_totsize => @mem_totsize, :mem_totfree => @mem_totfree }
         end
+
+        private
+
+        def show_deprecation_for_core_routes
+          return if request.path.starts_with?('/foreman_statistics')
+          Foreman::Deprecation.api_deprecation_warning('/api/v2/statistics API endpoint is deprecated, please use /foreman_statistics/api/v2/statistics instead')
```

Comment:
In Foreman you can also use the API path without `/v2`. You're not checking it explicitly. I also wonder if every plugin has its own root API. AFAIK Katello doesn't.

Reply:
Yeah, I'm not, you're right, I was not sure which to choose, but I thought this is enough to understand what's going on.
I'm just checking the prefix.
To the prefix, I've chosen to go with the prefix with this plugin along the way, I've written it in the blog post. There is really only one plugin doing that now I know of (`foreman_bootdisk`), but I believe it's a good practice as the plugin won't clash their urls/resources(classes, models, etc.)
I'd be happy to have that discussion if you'd like to though, I can see the downsides as well, as the plugin doesn't feel that integrated to the ecosystem as it would without the strict namespacing :)

Reply:
Perhaps we should get some feedback on discourse. Another thing to note is that it should be listed in the apidoc which is still shared. Otherwise tools like hammer and foreman-ansible-modules will not find them.

@evgeni I'm correct stating this for FAM, right?

Reply:
Things that are not in the apidoc don't exist ;)
So yeah, hammer/FAM can't use things that ate not exposed via apipie.

On the other hand we don't analyze any deprecations that might be noted there.

Reply:
Yeah, in the apidoc, if the resources would be named the same, they would clash. Currently we don't have a way around it I think. But if it would come to that, we can figure how to prefix the resources. Or give warnings about it at least, I think now on the clash we silently override the resource (that's what happens with this plugin, once it's installed, the `Trends` resource points to the plugins API endpoints)

Reply:
Yeah. Same happens with Katello which overrides the Organization resource (intentionally).